tests/Classifai/HelpersTest.php

```diff
@@ -135,4 +145,97 @@ function test_it_knows_configured_feature_taxonomies() {
 			$this->assertEquals( $taxonomy, $actual );
 		}
 	}
+
+	/**
+	 * @covers \Classifai\sort_images_by_size_cb
+	 */
+	public function test_sort_images_by_size_cb() {
+		$this->assertEquals(
+			0,
+			sort_images_by_size_cb(
+				[
+					'height' => 4,
+					'width'  => 6,
+				],
+				[
+					'height' => 2,
+					'width'  => 8,
+				]
+			)
+		);
+
+		$this->assertEquals(
+			-1,
+			sort_images_by_size_cb(
+				[
+					'height' => 4,
+					'width'  => 7,
+				],
+				[
+					'height' => 2,
+					'width'  => 8,
+				]
+			)
+		);
+
+		$this->assertEquals(
+			1,
+			sort_images_by_size_cb(
+				[
+					'height' => 4,
+					'width'  => 6,
+				],
+				[
+					'height' => 2,
+					'width'  => 9,
+				]
+			)
+		);
+	}
+
+	/**
+	 * @covers \Classifai\get_largest_acceptable_image_url
+	 */
+	public function test_get_largest_acceptable_image_url() {
+		$attachment = $this->factory->attachment->create_upload_object( DIR_TESTDATA .'/images/33772.jpg' ); // ~172KB image.
+
+		$set_150kb_max_filesize = function() {
+			return 150000;
+		};
+		add_filter( 'classifai_computer_vision_max_filesize', $set_150kb_max_filesize );
+
+		$url = get_largest_acceptable_image_url(
+			get_attached_file( $attachment ),
+			wp_get_attachment_url( $attachment, 'full' ),
+			wp_get_attachment_metadata( $attachment )['sizes'],
+			computer_vision_max_filesize()
+		);
+		$this->assertEquals( sprintf( 'http://example.org/wp-content/uploads/%s/%s/33772-1536x864.jpg', date( 'Y' ), date( 'm' ) ), $url );
+
+		$attachment = $this->factory->attachment->create_upload_object( DIR_TESTDATA .'/images/2004-07-22-DSC_0008.jpg' ); // ~109kb image.
+		$url = get_largest_acceptable_image_url(
+			get_attached_file( $attachment ),
+			wp_get_attachment_url( $attachment, 'full' ),
+			wp_get_attachment_metadata( $attachment )['sizes'],
+			computer_vision_max_filesize()
+		);
+		$this->assertEquals( sprintf( 'http://example.org/wp-content/uploads/%s/%s/2004-07-22-DSC_0008.jpg', date( 'Y' ), date( 'm' ) ), $url );
+
+		remove_filter( 'classifai_computer_vision_max_filesize', $set_150kb_max_filesize );
+
+		$set_1kb_max_filesize = function() {
+			return 1000;
+		};
+		add_filter( 'classifai_computer_vision_max_filesize', $set_1kb_max_filesize );
+
+		$url = get_largest_acceptable_image_url(
+			get_attached_file( $attachment ),
+			wp_get_attachment_url( $attachment, 'full' ),
+			wp_get_attachment_metadata( $attachment )['sizes'],
+			computer_vision_max_filesize()
+		);
+		$this->assertNull( $url );
+
+		remove_filter( 'classifai_computer_vision_max_filesize', $set_1kb_max_filesize );
```

Comment:
Small nitpick here, if the test fails, this filter won't be removed. I'd recommend moving it up before the `$this->assertNull` call.